service/collector/resource_group.go

```diff
@@ -0,0 +1,171 @@
+package collector
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sync/errgroup"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	labelID        = "id"
+	labelName      = "name"
+	labelState     = "state"
+	labelLocation  = "location"
+	labelManagedBy = "managed_by"
+)
+
+var (
+	resourceGroupDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName("azure_operator", "resource_group", "info"),
+		"Resource group information.",
+		[]string{
+			labelID,
+			labelName,
+			labelState,
+			labelLocation,
+			labelManagedBy,
+		},
+		nil,
+	)
+
+	gaugeValue float64 = 1
+)
+
+type ResourceGroupConfig struct {
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+
+	EnvironmentName string
+}
+
+type ResourceGroup struct {
+	k8sClient kubernetes.Interface
+	logger    micrologger.Logger
+
+	environmentName string
+}
+
+func NewResourceGroup(config ResourceGroupConfig) (*ResourceGroup, error) {
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.EnvironmentName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.EnvironmentName must not be empty", config)
+	}
+
+	r := &ResourceGroup{
+		k8sClient: config.K8sClient,
+		logger:    config.Logger,
+
+		environmentName: config.EnvironmentName,
+	}
+
+	return r, nil
+}
+
+func (r *ResourceGroup) Collect(ch chan<- prometheus.Metric) error {
+	clientSets, err := getUniqueClientSets(r.k8sClient, r.environmentName)
```

Comment:
And this way, we have a nice simple 'get all the clients, do a collect for each client' workflow like we have in the aws-operator :D